src/backend/distributed/metadata/dependency.c

```diff
@@ -776,15 +780,11 @@ DeferErrorIfHasUnsupportedDependency(const ObjectAddress *objectAddress)
 	#endif
 
 	/*
-	 * If the given object is a procedure or type, we want to create it locally,
-	 * so provide that information in the error detail.
+	 * Then we want to create it locally, so provide that information
+	 * in the error detail.
 	 */
-	if (getObjectClass(objectAddress) == OCLASS_PROC ||
-		getObjectClass(objectAddress) == OCLASS_TYPE)
-	{
-		appendStringInfo(detailInfo, "\"%s\" will be created only locally",
-						 objectDescription);
-	}
+	appendStringInfo(detailInfo, "\"%s\" will be created only locally",
```

Comment:
Since we'll always have detailInfo after that change, please update the lines below with ternary operator on detailInfo->data